posthog/temporal/workflows/logger.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import json
+import logging
+
+import aiokafka
+import structlog
+import temporalio.activity
+import temporalio.workflow
+from django.conf import settings
+from structlog.processors import EventRenamer
+from structlog.typing import FilteringBoundLogger
+
+from posthog.kafka_client.topics import KAFKA_LOG_ENTRIES
+
+
+async def bind_batch_exports_logger(team_id: int, destination: str | None = None) -> FilteringBoundLogger:
+    """Return a bound logger for BatchExports."""
+    if not structlog.is_configured():
+        await configure_logger()
+
+    logger = structlog.get_logger()
+
+    return logger.new(team_id=team_id, destination=destination)
+
+
+async def configure_logger(
+    logger_factory=structlog.PrintLoggerFactory,
+    extra_processors: list[structlog.types.Processor] | None = None,
+    queue: asyncio.Queue | None = None,
+    producer: aiokafka.AIOKafkaProducer | None = None,
+    cache_logger_on_first_use: bool = True,
+) -> tuple:
+    """Configure a StructLog logger for batch exports.
+
+    Configuring the logger involves:
+    * Setting up processors.
+    * Spawning a task to listen for Kafka logs.
+    * Spawning a task to shutdown gracefully on worker shutdown.
+
+    Args:
+        logger_factory: Optionally, override the logger_factory.
+        extra_processors: Optionally, add any processors at the end of the chain.
+        queue: Optionally, bring your own log queue.
+        producer: Optionally, bring your own Kafka producer.
+        cache_logger_on_first_use: Set whether to cache logger for performance.
+            Should always be True except in tests.
+    """
+    log_queue = queue if queue is not None else asyncio.Queue(maxsize=-1)
+    put_in_queue = PutInBatchExportsLogQueueProcessor(log_queue)
+
+    base_processors: list[structlog.types.Processor] = [
+        structlog.processors.add_log_level,
+        structlog.processors.format_exc_info,
+        structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f", utc=True),
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        add_batch_export_context,
```

Comment:
Could/should this work be done one time in `bind_batch_exports_logger` and added to the `logger.new`?

I guess it's premature optimization, it just seems expensive to do per line and it doesn't seem like anything changes?

Reply:
Yeah, this could be bound instead.